pil-analyzer/src/statement_processor.rs

```diff
@@ -428,18 +445,56 @@ where
                     length: None,
                     degree: None,
                 };
-                let value = FunctionValueDefinition::TypeConstructor(
+                let value = FunctionValueDefinition::TypeConstructor(TypeConstructor::Enum(
                     shared_enum_decl.clone(),
                     variant.clone(),
-                );
+                ));
                 PILItem::Definition(var_symbol, Some(value))
             });
             return iter::once(PILItem::Definition(
                 symbol,
-                Some(FunctionValueDefinition::TypeDeclaration(enum_decl.clone())),
+                Some(FunctionValueDefinition::TypeDeclaration(
+                    TypeDeclarationAnalyzed::Enum(enum_decl.clone()),
+                )),
             ))
             .chain(var_items)
             .collect();
+        } else if let Some(FunctionDefinition::TypeDeclaration(TypeDeclarationParsed::Struct(
+            struct_decl,
+        ))) = value
+        {
+            assert_eq!(symbol_kind, SymbolKind::Other());
+            let struct_decl = self.process_struct_declaration(struct_decl);
+            let shared_struct_decl = Arc::new(struct_decl.clone());
+            let field_items = struct_decl.fields.keys().map(|field_name| {
```

Comment:
I don't think the field items should be accessible through the type. Are they in rust?